internal/collectors/virtualization.go

```diff
@@ -0,0 +1,34 @@
+package collectors
+
+import (
+	"fmt"
+
+	"github.com/SUSE/connect-ng/internal/util"
+)
+
+type Virtualization struct{}
+
+func (Virtualization) run(arch Architecture) (Result, error) {
+
+	// We utilize systemd here to fetch the virtualization information. Since we do not hard require systemd
+	// there might be the possibility to run on a system without systemd installed (e.g. containers).
+	// This tool can detect a lot different hypervisors:
+	// https://github.com/systemd/systemd/blob/main/src/basic/virt.c#L1046
+	if !util.ExecutableExists("systemd-detect-virt") {
+		return NoResult, fmt.Errorf("can not detect virtualization: `system-detect-virt` executable not found")
```

Comment:
Small typo here:

```suggestion
		return NoResult, fmt.Errorf("can not detect virtualization: `systemd-detect-virt` executable not found")
```

By the way, do you think it's worth having `systemd-detect-virt` in a constant? It is used in at least three different spots.

Reply:
I have no opinion but I will move it to a constant :)

Reply:
moved it!